source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ViewController.m

```diff
@@ -107,11 +142,48 @@ - (void)update:(NSString *) jsonStr
         adcVc.acrActionDelegate = self;
         if(self.curView)
             [self.curView removeFromSuperview];
+        else
+        {
+            self.scrView = [[UIScrollView alloc] initWithFrame:CGRectMake(0,0,0,0)];
+            self.scrView.showsHorizontalScrollIndicator = YES;
+        }
         self.curView = adcVc.view;
-        self.curView.frame = CGRectMake(20, 250, 300, 1250);
+        self.scrView.translatesAutoresizingMaskIntoConstraints = NO;
+        NSLog(@"ContentView H = %f", self.curView.frame.size.height);
+        NSLog(@"ContentView W = %f", self.curView.frame.size.width);
+        
+        //self.curView.frame = CGRectMake(0, 0, 0, 0);
         [self addChildViewController:adcVc];
-        [self.view addSubview:adcVc.view];
+        [self.scrView addSubview:adcVc.view];
         [adcVc didMoveToParentViewController:self];
+        self.scrView.contentSize = self.curView.frame.size;
+        NSLog(@"ContentView H = %f", self.scrView.contentSize.height);
+        NSLog(@"ContentView W = %f", self.scrView.contentSize.width);
+        
+        UIScrollView *scrollview = self.scrView;
+        UIView *view = self.curView;
+        view.translatesAutoresizingMaskIntoConstraints = NO;
+        scrollview.translatesAutoresizingMaskIntoConstraints = NO;
+        
+        NSDictionary *viewMap = NSDictionaryOfVariableBindings(view, scrollview);
+        NSArray<NSString *> *formats =
+        [NSArray arrayWithObjects:@"H:|-10-[view(<=scrollview)]-10-|", @"V:|-[view(>=scrollview)]-|",nil];
+        NSArray<NSLayoutConstraint *> *constraints = nil;
+        
+        
+        for(NSString *format in formats)
+        {
+            constraints = [NSLayoutConstraint constraintsWithVisualFormat:format
+                                                                  options:0
+                                                                  metrics:nil
+                                                                    views:viewMap];
+            for(NSLayoutConstraint *con in constraints)
+            {
+                con.active = YES;
+            }
```

Comment:
You can use `activateConstraints` here as well.

Reply:
thanks

---
In reply to: [143583803](https://github.com/Microsoft/AdaptiveCards/pull/760#discussion_r143583803) [](ancestors = 143583803)